[Testing] pipe_round_robin.swift.gyb is scheduling sensitive

davezarzycki opened the pull request:
If a background compile job is running on my Linux box, this test will sometimes fail. Also, let's up the buffer size while we're in here. One kilobyte is needlessly tiny.

davezarzycki commented:
@swift-ci please smoke test

davezarzycki commented:
@swift-ci please smoke test linux

davezarzycki commented:
@swift-ci please smoke test

davezarzycki commented:
@swift-ci please clean test linux

davezarzycki commented:
@swift-ci please smoke test macos

davezarzycki commented:
@graydon – Might you please review this? Thanks!

graydon reviewed (changes requested):
While I appreciate that increasing the buffer size makes the buffered reading more efficient, it also potentially defeats the way this test works, since we're testing the read-to-poll ratio and each read now consumes 64x as much data.

Can you check to be sure that the read-to-poll ratio is still very high (hundreds-to-thousands) when running the preimage of the commit this test was added for, with an increased read-buffer size? Otherwise the test is no longer meaningful as-is, and needs some further refinement.

davezarzycki committed:
[Testing] pipe_round_robin.swift.gyb is scheduling sensitive

If a background compile job is running on my Linux box, this test will
sometimes fail.

davezarzycki commented:
I don't want to get sucked into cleaning up `lib/Basic/Unix/TaskQueue.inc`, so I removed the performance improvement.

@swift-ci please smoke test

graydon commented:
Looks good, thanks!
